File: Sources/CountriesFlagsObjC/source/Countries/UIImage+Latvia.m
//
//  UIImage+Latvia.m
//  CountriesFlags
//

#import "Countries/UIImage+Latvia.h"

@implementation UIImage (Latvia)

+ (UIImage * _Nullable)flagLatviaWithSize:(CGSize)size {
    UIColor* colorRed = [UIColor colorWithRed:157.0f/255.0f
                                        green:34.0f/255.0f
                                         blue:53.0f/255.0f
                                        alpha:1.0f];
    UIColor* colorWhite = [UIColor whiteColor];
    
    UIGraphicsBeginImageContextWithOptions(size, NO, 0.0f);
    
    [colorRed setFill];
    UIRectFill(CGRectMake(0.0f, 0.0f, size.width, size.height));
    
    [colorWhite setFill];
    UIRectFill(CGRectMake(0.0f, size.height / 5.0f * 2.0f, size.width, size.height / 5.0f));
    
    UIImage* image = UIGraphicsGetImageFromCurrentImageContext();
    UIGraphicsEndImageContext();
    
    return image;
}

@end

File: Sources/CountriesFlags/Countries/UIImage+Latvia.swift
//
//  UIImage+Latvia.swift
//  CountriesFlags
//

#if canImport(UIKit)

import UIKit

extension UIImage {
    /**
     [wikipedia]: https://en.wikipedia.org/wiki/Flag_of_Latvia "wikipedia"
     
     Generates the flag of Latvia.
     For more information see [wikipedia].

     - Parameters:
        - size: image size.
     
     - Returns: Returns the flag of Latvia.
     */
    internal static func flagLatvia(with size: CGSize) -> UIImage {
        let colorWhite = UIColor.white
        let colorRed = UIColor(red: 157.0 / 255.0,
                               green: 34.0 / 255.0,
                               blue: 53.0 / 255.0,
                               alpha: 1.0)

        UIGraphicsBeginImageContextWithOptions(size, true, 0.0)

        colorRed.setFill()
        UIRectFill(CGRect(origin: .zero, size: size))

        colorWhite.setFill()
        UIRectFill(CGRect(origin: CGPoint(x: 0.0, y: size.height / 5.0 * 2.0),
                          size: CGSize(width: size.width, height: size.height / 5.0)))

        let image = UIGraphicsGetImageFromCurrentImageContext()
        UIGraphicsEndImageContext()

        return image ?? UIImage()
    }
}

#endif
